src/sqlfluff/core/linter/linter.py

```diff
@@ -514,7 +518,17 @@ def lint_fix_parsed(
                 if fix and fixes:
                     linter_logger.info(f"Applying Fixes [{crawler.code}]: {fixes}")
                     # Do some sanity checks on the fixes before applying.
-                    if fixes == last_fixes:  # pragma: no cover
+                    # :TRICKY: Use IdentitySet rather than set() since
+                    # different segments may compare as equal.
+                    unique_anchors = IdentitySet(fix.anchor for fix in fixes)
+                    if len(unique_anchors) < len(fixes):  # pragma: no cover
+                        message = (
+                            f"Rule {crawler.code} returned multiple fixes with "
+                            f"the same anchor. This is not supported, so the "
+                            f"fixes will not be applied. {fixes!r}"
+                        )
+                        cls._warn_duplicate_anchors(message)
```

Comment:
I presume this has to be unique per rule?

i.e. if you have this:

```sql
SELECT DATE_ADD(date,col1)
```

And you had to change that to:

```sql
SELECT DATE_ADD(date, "col1")
```

That is add a space AND change `col1` to be quoted, would that be same anchor point? But as it's different rules there shouldn't be a problem? Or do all rules fixes get queued up to one big set?

I know you say two fixes for same anchor points don't work, but just wondering if they do in some cases and we're now going to break that?

Should this throw a CRITICAL exception so my rules checker will see that?

Reply:
Unique per rule, yes. The linter runs one rule at a time, applies all the fixes, then runs the next rule, and this check occurs within each iteration of the rule loop.

I could change the log level if you like. This is similar to the recent "parse after fix", which currently log warnings. If we change this, should we consider changing that as well?

Reply:
I think I need to update my script to look for more than just CRITICAL log lines if we want to catch these things. Otherwise they’re silently hidden and we only get notice when users raise them.

Or maybe we should consider them critical?

We could also use that key “CRITICAL” word in the log, so my script catches, but then a bit weird that it’s just a warning log level if we do that. It mightn’t be a bad idea to do that before merging, let one run of CI go through, then revert it back assuming no issues. Longer term I should just fix my script though to look for more key phrases we want it to catch.

Reply:
I'm fine with elevating it to CRITICAL. Since these are bugs in the rules, I don't mind drawing more attention to this.

Reply:
Cool let’s do that and see if my script picks up anything!

Reply:
Changing this now.